packages/zendesk-adapter/src/change_validators/guide_order/article_order_validator.ts

```diff
@@ -14,11 +14,14 @@
 * limitations under the License.
 */
 import { ChangeValidator } from '@salto-io/adapter-api'
-import { validateReferences } from './guide_order_validators_utils'
-import { ARTICLES_FIELD, ARTICLE_ORDER_TYPE_NAME } from '../../constants'
+import { validateOrderElementAdded, validateReferences } from './guide_order_validators_utils'
+import { ARTICLES_FIELD, ARTICLE_ORDER_TYPE_NAME, ARTICLE_TYPE_NAME } from '../../constants'
 
 /**
  * Validates that all the elements in the articles order list are references
  */
 export const articleOrderValidator: ChangeValidator = async changes =>
-  validateReferences(changes, ARTICLES_FIELD, ARTICLE_ORDER_TYPE_NAME)
+  [
+    ...validateReferences(changes, ARTICLES_FIELD, ARTICLE_ORDER_TYPE_NAME),
+    ...validateOrderElementAdded(changes, ARTICLES_FIELD, ARTICLE_ORDER_TYPE_NAME, ARTICLE_TYPE_NAME),
+  ]
```

Comment:
there is a limit to concatenating long arrays - please use `[].concat()` or something similar instead of `...` (this will crash on large accounts)
```suggestion
  validateReferences(changes, ARTICLES_FIELD, ARTICLE_ORDER_TYPE_NAME).concat(
    validateOrderElementAdded(changes, ARTICLES_FIELD, ARTICLE_ORDER_TYPE_NAME
  )
```

Reply:
Are you sure? i've managed to do it on 16 million items in the developer tools console, without any issue

Reply:
yes, we've had issues with this many times in the past...